nullability/v0.2/nullability-v0.2.graphql

```diff
@@ -0,0 +1,100 @@
+"""
+Indicates that a position is semantically non null: it is only null if there is a matching error in the `errors` array.
+In all other cases, the position is non-null.
+
+Tools doing code generation may use this information to generate the position as non-null:
+
+```graphql
+type User {
+    # email is semantically non-null and can be generated as non-null by error-handling clients.
+    email: String @semanticNonNull
+}
+```
+
+The `field` argument is the name of the field if `@semanticNonNull` is applied to an object type extension:
+
+```graphql
+# extend the base schema to make User.email semantically non-null.
+extend type User @semanticNonNull(field: "email")
+```
+
+`field` is used for client applications that do not control the base schema and must use type extensions.
+When used on a field definition, `field` must be null.
+
+The `level` argument indicates what level is semantically non null in case of lists:
+
+```graphql
+type User {
+    # friends is semantically non null
+    friends: [User] @semanticNonNull # same as @semanticNonNull(level: 0)
+
+    # friends[0], ... are semantically non null
+    friends: [User] @semanticNonNull(level: 1)
+
+    # friends, friends[0], ... are all semantically non null
+    friends: [User] @semanticNonNull(level: 0) @semanticNonNull(level: 1)
+}
+```
+
+`level` is zero indexed.
+Passing a negative level or a level greater than the list dimension is an error.
```

Comment:
Presumably it's also an error to define level at all on a non-list field? Maybe there's a way to word this which covers both cases?

Reply:
I would vote for having `@semanticNonNull` have exactly the same meaning as `@semanticNonNull(level: 0)`? If there's no list then the list dimension is 0, which is also the default value for the argument? 
Both would be allowed on a non-list field.

Reply:
Sounds reasonable!